src/init.cpp

```diff
@@ -462,8 +462,8 @@ void SetupServerArgs(ArgsManager& argsman)
     argsman.AddArg("-prune=<n>", strprintf("Reduce storage requirements by enabling pruning (deleting) of old blocks. This allows the pruneblockchain RPC to be called to delete specific blocks and enables automatic pruning of old blocks if a target size in MiB is provided. This mode is incompatible with -txindex. "
             "Warning: Reverting this setting requires re-downloading the entire blockchain. "
             "(default: 0 = disable pruning blocks, 1 = allow manual pruning via RPC, >=%u = automatically prune block files to stay under the specified target size in MiB)", MIN_DISK_SPACE_FOR_BLOCK_FILES / 1024 / 1024), ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
-    argsman.AddArg("-reindex", "Rebuild chain state and block index from the blk*.dat files on disk. This will also rebuild active optional indexes.", ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
-    argsman.AddArg("-reindex-chainstate", "Rebuild chain state from the currently indexed blocks. When in pruning mode or if blocks on disk might be corrupted, use full -reindex instead.", ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
+    argsman.AddArg("-reindex", "Rebuild chain state and block index from the blk*.dat files on disk. This will also rebuild active optional indexes. Warning: Not yet validated snapshot chainstates will be removed.", ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
+    argsman.AddArg("-reindex-chainstate", "Rebuild chain state from the currently indexed blocks. When in pruning mode or if blocks on disk might be corrupted, use full -reindex instead. Warning: Not yet validated snapshot chainstates will be removed.", ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
```

Comment:
In commit "doc: Add snapshot chainstate removal warning to reindexing documentation" (2bd1d39491f8ad811dcb6d3787e1e18ca0b928ad)

These warnings seem too alarmist, and also not very actionable. The point of reindex commands is to wipe state and rebuild it. So it should not be surprising that state loaded from an assumeutxo snapshot is also wiped. I think it would be a better to add a clarification instead of a warning here, and a suggestion to reload the snapshot. Maybe: "If an assumeutxo snapshot was loaded and not yet validated, it will need to be reloaded for its chain state to be rebuilt."

Reply:
Done, adapted with your text.